spec/ntheory_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'spec_helper'
+
+describe SymEngine do
+  before :each do
+  end
+
+  describe 'NTheory' do
+
+    describe '#gcd' do
+      context 'GCD of 2 and 4' do
+        it 'returns 2' do
+          f = SymEngine::gcd(2, 4)
+          expect(f).to eql(2)
+        end
+      end
+    end
+
+    describe '#lcm' do
+      context 'LCM of 2 and 4' do
+        it 'returns 4' do
+          f = SymEngine::lcm(2, 4)
+          expect(f).to eql(4)
+        end
+      end
+    end
+
+    describe '#nextprime' do
+      context 'NextPrime after 4' do
+        it 'returns 5' do
+          f = SymEngine::nextprime(4)
+          expect(f).to eql(5)
+        end
+      end
+    end
+
+    describe '#mod' do
+
+      i4 = SymEngine::Integer.new(4)
+      i5 = SymEngine::Integer.new(5)
+      i_4 = SymEngine::Integer.new(-4)
+      i_5 = SymEngine::Integer.new(-5)
+
+      context '5 mod 4' do
+        it 'returns 1' do
+          f = i5 % i4
+          expect(f).to eql(1)
+        end
+      end
+      context '-5 mod -4' do
+        it 'returns -1' do
+          f = i_5 % i_4
+          expect(f).to eql(-1)
+        end
+      end
+      context '5 mod -4' do
+        it 'returns -3' do
+          f = i5 % i_4
+          expect(f).to eql(-3)
+        end
+      end
+      context '-5 mod 4' do
+        it 'returns 3' do
+          f = i_5 % i4
+          expect(f).to eql(3)
+        end
+      end
+    end
+
+    describe '#quotient' do
+      context 'quotient of 5 divided by 2' do
+        it 'returns 2' do
+          f = SymEngine::quotient(5, 2)
+          expect(f).to eql(2)
+        end
+      end
+    end
+
+    describe '#fibonacci' do
+      context '5th Fibonacci Number' do
+        it 'returns 5' do
+          f = SymEngine::fibonacci(5)
```

Comment:
Better to add it to all of them. Or at least one for one argument and one for two argument function.